persistence/src/main/java/edu/unc/lib/dl/persist/services/move/MoveObjectsJob.java

```diff
@@ -141,10 +143,63 @@ private void moveObject(PID objPid) {
         PID sourcePid = moveContent.getParent().getPid();
         addPidToSource(objPid, sourcePid);
 
+        // Write a premis event if object moves between admin units
+        adminUnitMove(objPid, moveContent);
+
         // Add the object to its destination, which clears the previous parent as well
         destContainer.addMember(moveContent);
     }
 
+    private void adminUnitMove(PID sourcePid, ContentObject moveObj) {
+        HashMap<String, String> destContainerInfo = getContainerInfo(destinationPid, 1);
+        String destAdminUnit = destContainerInfo.get("adminUnit");
+
+        HashMap<String, String> currentContainerInfo = getContainerInfo(sourcePid, 2);
+        String currentAdminUnit = currentContainerInfo.get("adminUnit");
+
+        if (currentAdminUnit != null && destAdminUnit != null && !currentAdminUnit.equals(destAdminUnit)) {
+            moveObj.getPremisLog().buildEvent(Premis.MetadataModification)
+                    .addAuthorizingAgent(agent.getUsername())
+                    .addEventDetail("Object moved from source {0} ({1}) in Admin Unit {2} ({3}) " +
+                                    "to destination {4} ({5}) in Admin Unit {6} ({7})",
+                            currentContainerInfo.get("container"), currentContainerInfo.get("containerTitle"),
+                            currentAdminUnit, currentContainerInfo.get("adminUnitTitle"),
+                            destContainerInfo.get("container"), destContainerInfo.get("containerTitle"),
+                            destAdminUnit, destContainerInfo.get("adminUnitTitle"))
+                    .writeAndClose();
+        }
+    }
+
+    private HashMap<String, String> getContainerInfo(PID pid, int entry) {
```

Comment:
```suggestion
    private Map<String, String> getContainerInfo(PID pid, int entry) {
```

Also, could you add javadoc to this? Particularly it'd be helpful to describe what `entry` is since its a bit tricky that its actually counting backwards